var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -131,61 +141,78 @@ def headers(self):
         lapacke_h = join_path(include_dir, 'lapacke.h')
         return HeaderList([cblas_h, lapacke_h])
 
-    def install_one(self, spec, prefix, shared):
-        cmake_args = [
-            '-DBUILD_SHARED_LIBS:BOOL=%s' % ('ON' if shared else 'OFF'),
-            '-DCMAKE_BUILD_TYPE:STRING=%s' % (
-                'Debug' if '+debug' in spec else 'Release'),
+    @property
+    def build_directory(self):
+        return join_path(self.stage.source_path,
+                         'spack-build-shared' if self._building_shared
+                         else 'spack-build-static')
+
+    def cmake_args(self):
+        args = [
+            '-DBUILD_SHARED_LIBS:BOOL=%s' % (
+                'ON' if self._building_shared else 'OFF'),
             '-DLAPACKE:BOOL=%s' % (
-                'ON' if '+lapacke' in spec else 'OFF'),
+                'ON' if '+lapacke' in self.spec else 'OFF'),
             '-DLAPACKE_WITH_TMG:BOOL=%s' % (
-                'ON' if '+lapacke' in spec else 'OFF')]
-        if spec.satisfies('@3.6.0:'):
-            cmake_args.extend(['-DCBLAS=ON'])  # always build CBLAS
+                'ON' if '+lapacke' in self.spec else 'OFF')]
+
+        if self.spec.satisfies('@3.6.0:'):
+            args.extend(['-DCBLAS=ON'])  # always build CBLAS
```

Comment:
Can you use `append()` instead of `extend([])`? Same elsewhere. No need to create a length one list and immediately destroy it.

Reply:
I didn't want to touch the code that I didn't need to touch. But OK.